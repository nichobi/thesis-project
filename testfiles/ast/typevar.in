[T-True]
True : Bool

[T-False]
False : Bool

[T-Or]
a : Bool,
b : Bool
---------------
Or(a, b) : Bool

[T-Zero]
Zero : Int

[T-IfElse]
x: Bool,
y: a,
z: a
------------------
IfElse(x, y, z): a

