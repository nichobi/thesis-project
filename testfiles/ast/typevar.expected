RuleSet
  List
    Rule Name="T-True"
      HasType
        Value ID="True"
        TyVal ID="Bool"
      List
    Rule Name="T-False"
      HasType
        Value ID="False"
        TyVal ID="Bool"
      List
    Rule Name="T-Or"
      HasType
        Function ID="Or"
          List
            Value ID="a"
            Value ID="b"
        TyVal ID="Bool"
      List
        HasType
          Value ID="a"
          TyVal ID="Bool"
        HasType
          Value ID="b"
          TyVal ID="Bool"
    Rule Name="T-Zero"
      HasType
        Value ID="Zero"
        TyVal ID="Int"
      List
    Rule Name="T-IfElse"
      HasType
        Function ID="IfElse"
          List
            Value ID="x"
            Value ID="y"
            Value ID="z"
        TyVar ID="a"
      List
        HasType
          Value ID="x"
          TyVal ID="Bool"
        HasType
          Value ID="y"
          TyVar ID="a"
        HasType
          Value ID="z"
          TyVar ID="a"
